addons/pos_cache/models/pos_cache.py

```diff
@@ -25,8 +25,9 @@ def refresh_all_caches(self):
 
     def refresh_cache(self):
         for cache in self:
-            Product = self.env['product.product'].with_user(cache.compute_user_id.id)
+            Product = self.env['product.product'].with_company(cache.config_id.company_id).with_user(cache.compute_user_id.id)
             products = Product.search(cache.get_product_domain())
+            products.invalidate_cache()
```

Comment:
why ?

Reply:
because for some reason in the loop, however the with_user and with_company, the cache is not consistent with the working company. Except by doing an invalidate